ThirdPartyAdapters/unity/unity/src/main/java/com/google/ads/mediation/unity/UnityAdapter.java

```diff
@@ -282,34 +246,57 @@ public void requestBannerAd(
   @Override
   public void onDestroy() {
     mMediationInterstitialListener = null;
-    if (bannerAd == null) {
-      return;
+    if (bannerAd != null) {
+      bannerAd.onDestroy();
     }
-    bannerAd.onDestroy();
   }
 
   @Override
   public void onPause() {
-    if (bannerAd == null) {
-      return;
+    if (bannerAd != null) {
+      bannerAd.onPause();
     }
-    bannerAd.onPause();
   }
 
   @Override
   public void onResume() {
-    if (bannerAd == null) {
-      return;
+    if (bannerAd != null) {
+      bannerAd.onResume();
     }
-    bannerAd.onResume();
   }
 
   @Override
   public View getBannerView() {
-    if (bannerAd == null) {
-      return null;
+    if (bannerAd != null) {
+      return bannerAd.getBannerView();
+    }
+    return null;
+  }
+
+  private void sendInterstitialAdEvent(AdEvent adEvent) {
+    if (mMediationInterstitialListener == null) {
```

Comment:
there is a check for null in many places but the property is not marked as weak reference. Just wondering if retain cycle is possible here?

Reply:
I do feel your pain here ;) However, from my understanding in Java its managed by garbage collector that works slightly different from Reference Counting.

Reply:
Actually, it has nothing to do with using weak references or not here. It's a safeguard since the listener is provided by the mediation SDK and we don't control this part. Theoretically, it should never be `null` but this is to protect ourselves from NPE.